### ISSUE TITLE
System Desc Slicing

### ISSUE THREAD

**Issue by nsmithtt** [tenstorrent/tt-mlir#113]
In order to make our runtime more flexible I would like to introduce the concept of system desc slicing.

For example, on a t3000 machine, today the user could do:
```
ttrt query --save-system-desc t3000.ttsys
```

But perhaps the user has many models they wish to run in parallel and resource share the devices on the system.  System desc slicing or partitioning (whatever we want to call it) would enable the user to do something like:

```
ttrt query --save-system-desc --devices 4,5,6 3device_system.ttsys
```

This would take the whole system descriptor, carve out 3 devices from it, and generate a new system descriptor that is effectively a view of *only* those 3 devices on the system.  If this `3device_system.ttsys` gets loaded into the compiler, the compiler will compile as though it were targeting a 3 device system and has no idea that this originally came from a t3000.

This gets way more complicated on galaxy, so let's just focus on t3000 like systems for now, but keep in mind that slicing up galaxy systems might be an interesting use case in the future.

**Comment by staylorTT** [comment]
@tapspatel Is this still an active issue?

**Comment by tapspatel** [comment]
its a nice to have, but its not being worked on.
